### PR TITLE
plotting|util|tests: Fix and test re-trying of plots which failed to load

### LINKED ISSUE #8262: [BUG] Harvester plots_refresh_parameter not keeping track of plots
The harvester is losing track of my plots during replotting, and my plot count is going down.

I have 5156 plots. All of my drives are full, so I am currently replotting.
I have a script that, upon completion of a new NFT plot, it identifies an OG plot, deletes it, and moves the new NFT plot in its place.
I am currently creating 64 plots every 24h.
Earlier today, my harvester did indeed show that it was farming `5156` plots. I check my status a few moments ago, and it shows that I'm only farming `5129` plots.

(Then some time passes for the farmer to rediscover all my plots)

Ok, so back to `5156` after restarting the farmer. It lost track of 27 plots.

When I first discovered this bug this morning, there was a Δ of 85 plots that the harvester wasn't tracking.

> 2021-08-28T13:54:20.544 harvester chia.harvester.harvester: INFO  ...  Total 5071 plots

`chia stop all -d`
`chia start farmer`

> 2021-08-28T14:49:27.451 harvester chia.harvester.harvester: INFO  ...  Total 5156 plots

What I suspect:
My config.yaml had been with me since before launch, and I decided to compare it against https://github.com/Chia-Network/chia-blockchain/blob/main/chia/util/initial-config.yaml to see if I needed to refresh it any. There were a few things that I was missing.
I had `plot_loading_frequency_seconds: 120`, which isn't anywhere in the current initial-config.yaml, so I swapped that out with: https://github.com/Chia-Network/chia-blockchain/blob/main/chia/util/initial-config.yaml#L128-L131
I think that this may not be discovering new plots for some reason. I'm going to attempt to remove those lines, and go back to `plot_loading_frequency_seconds: 120`. I'll check back in the morning to see if the count is off.

Chia: 1.2.5
Distro: Arch Linux [x86_64]
Kernel: Linux 5.13.12-arch1-1
Processor: 24 x Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
Processor: 24 x Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
RAM: 64362 MiB

### PR DESCRIPTION
**Pull request by xdustinface**
Should fix #8262

### CODE REVIEW

File: chia/plotting/manager.py [hunk 1]
```diff
@@ -162,7 +162,8 @@ def process_file(file_path: Path) -> Dict:
             if file_path.exists():
                 if (
                     file_path in self.failed_to_open_filenames
-                    and (time.time() - self.failed_to_open_filenames[file_path]) > 1200
+                    and (time.time() - self.failed_to_open_filenames[file_path])
+                    < self.refresh_parameter.retry_invalid_seconds
                 ):
                     # Try once every 20 minutes to open the file
```
Comment (arvidn) [inline chia/plotting/manager.py:168]: this comment is outdated with this patch

File: chia/plotting/manager.py [hunk 2]
```diff
@@ -162,7 +162,8 @@ def process_file(file_path: Path) -> Dict:
             if file_path.exists():
                 if (
                     file_path in self.failed_to_open_filenames
-                    and (time.time() - self.failed_to_open_filenames[file_path]) > 1200
+                    and (time.time() - self.failed_to_open_filenames[file_path])
+                    < self.refresh_parameter.retry_invalid_seconds
```
Comment (arvidn) [inline chia/plotting/manager.py:166]: this is the bug fix, as I understand it. Flipping `>` to `<`

File: chia/plotting/manager.py [hunk 3]
```diff
@@ -268,6 +269,9 @@ def process_file(file_path: Path) -> Dict:
                         result.loaded_plots += 1
                         result.loaded_size += stat_info.st_size
 
+                    if file_path in self.failed_to_open_filenames:
+                        del self.failed_to_open_filenames[file_path]
```
Comment (arvidn) [inline chia/plotting/manager.py:273]: this is to remove files that eventually *succeed* from the failed-list, right?

**Review by arvidn — DISMISSED** [review]
in an ideal world, that function would be a lot shorter, and logic from it would be factored out and unit tested separately (and without using wall clock time).

I understand you made this timeout configurable in order to allow a test to take less than 20 minutes to run, but still takes 15 seconds or so

File: chia/plotting/manager.py (hunk 1 above)
Comment (xdustinface) [reply chia/plotting/manager.py]: Right, changed!

File: chia/plotting/manager.py (hunk 2 above)
Comment (xdustinface) [reply chia/plotting/manager.py:166]: Correct

File: chia/plotting/manager.py (hunk 3 above)
Comment (xdustinface) [reply chia/plotting/manager.py:273]: Also correct.

**Review by arvidn — APPROVED** [review]

**Pull request merged**